File: src/main/java/FlyAway/reservation/dto/ReservationSummaryClientDto.java
package FlyAway.reservation.dto;

import FlyAway.flight.aircraft.CabinClass;
import FlyAway.flight.dto.FlightBasicDto;
import FlyAway.reservation.ReservationStatus;

import java.util.UUID;

public record ReservationSummaryClientDto(
        UUID id,
        ReservationStatus status,
        CabinClass cabinClass,
        FlightBasicDto flightDto

) {
}

File: frontend/src/app/services/models/reservation-summary-client-dto.ts
/* tslint:disable */
/* eslint-disable */
import { FlightBasicDto } from '../models/flight-basic-dto';
export interface ReservationSummaryClientDto {
  cabinClass?: 'FIRST' | 'BUSINESS' | 'ECONOMY';
  flightDto?: FlightBasicDto;
  id?: string;
  status?: 'PENDING' | 'ACTIVE' | 'CANCELLED' | 'EXPIRED' | 'COMPLETED' | 'FAILED';
}
